src/Microsoft.Framework.Logging.Console/ConsoleLoggerFactoryExtensions.cs

```diff
@@ -35,5 +37,23 @@ public static ILoggerFactory AddConsole(this ILoggerFactory factory, LogLevel mi
             factory.AddProvider(new ConsoleLoggerProvider((category, logLevel) => logLevel >= minLevel));
             return factory;
         }
+
+        /// <summary>
+        ///     Adds a console logger that is enabled as defined by the IConfiguration object
+        /// </summary>
+        /// <param name="configuration">IConfiguration object with filter attributes</param>
+        /// <returns></returns>
+        public static ILoggerFactory AddConsole(this ILoggerFactory factory, IConfiguration configuration)
+        {
+            factory.AddProvider(new ConsoleLoggerProvider((category, logLevel) =>
+            {
+                LogLevel minLevel;
+
+                return Enum.TryParse(configuration?.GetSubKey("ConsoleLogger")?.Get(category), out minLevel) &&
```

Comment:
It returns a `IEnumerable<KeyValuePair<string,IConfiguration>>` so we have iterate of each object to read the key value. But I think this confuses me in the sense that the value in KeyValuePair is IConfiguration, so entries of type "category":"value" would be cast to IConfiguration ? Let me try out a small example and get back